oximeter/db/src/client.rs

```diff
@@ -106,6 +111,15 @@ impl Client {
             }
             None => BTreeMap::new(),
         };
+
+        // Error if a limit is specified with a query that resolves to multiple
+        // timeseries. Because query results are ordered in part by timeseries
+        // key, results with a limit will selected measurements in a way that is
+        // arbitrary with respect to the query.
+        if limit.is_some() && info.len() != 1 {
+            return Err(Error::InvalidLimitQuery);
+        }
+
```

Comment:
@bnaecker I erred on the side of erring here even when the number of queries is zero, on the idea that this error is more about character of the request, and if there are zero results that's probably not what the caller wanted. However, I could see it either way.

Reply:
Ok, I talked myself out of it. Changed in 6e897d1503e23110de5aee18cb66e7110f31bae9

Reply:
Ok, either one sounds reasonable to me. Thanks for making that change here!